test/toJSON.test.js

```diff
@@ -170,5 +172,37 @@ test('throw an error on non nullable null object', (t) => {
       }
     }
   })
-  t.throws(() => { stringify(null) })
+
+  const result = stringify(null)
+  t.equal(result, '{}')
+})
+
+test('on non-nullable null object with required fields it should throw complaining missing required fields', (t) => {
```

Comment:
I didn't really think about this beforehand, but seeing this happening I thought it could make sense to have this test.